spec/controllers/api/v1/books_controller_spec.rb

```diff
@@ -0,0 +1,77 @@
+require 'rails_helper'
+
+RSpec.describe Api::V1::BooksController, type: :controller do
+
+  describe 'GET #index' do
+    subject(:http_request) { get :index }
+    let!(:books) { create_list(:book, 3) }
+
+    context 'When user is authenticated' do
+      include_context 'Authenticated User'
+
+      before do
+        http_request
+      end
+
+      context 'When fetching all the books' do
+
+        it 'paginates records' do
+          expect(JSON.parse(response.body)).to be_paginated
+        end
+
+        it 'is serialized with BookSerializer' do
+          expect(JSON.parse(response.body)['page']).to have_been_serialized_with(BookSerializer)
+        end
+
+        it 'includes all book ids' do
+          expected_book_ids = books.pluck(:id)
+          response_book_ids = JSON.parse(response.body)['page'].map {|b| b['id']}
+          expect(expected_book_ids.sort).to eql(response_book_ids.sort)
```

Comment:
The approach I usually use for this kind of tests consist in using the [`all?` method](https://apidock.com/ruby/Enumerable/all%3F) along with `include?`. Something like:
```ruby
expect (response_book_ids.all? { |book_id|  expected_book_ids.include?(book_id) }).to 
be_truthy
```

Note: This is not a requested change, but something to bear in mind :)